src/main/java/bio/terra/workspace/service/crl/CrlService.java

```diff
@@ -0,0 +1,116 @@
+package bio.terra.workspace.service.crl;
+
+import bio.terra.cloudres.common.ClientConfig;
+import bio.terra.cloudres.common.cleanup.CleanupConfig;
+import bio.terra.cloudres.google.billing.CloudBillingClientCow;
+import bio.terra.cloudres.google.cloudresourcemanager.CloudResourceManagerCow;
+import bio.terra.cloudres.google.serviceusage.ServiceUsageCow;
+import bio.terra.workspace.app.configuration.external.CrlConfiguration;
+import bio.terra.workspace.service.crl.exception.CrlInternalException;
+import bio.terra.workspace.service.crl.exception.CrlNotInUseException;
+import bio.terra.workspace.service.crl.exception.CrlSecurityException;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.time.Duration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CrlService {
+  /** The client name required by CRL. */
+  public static final String CLIENT_NAME = "workspace";
+  /** How long to keep the resource before Janitor do the cleanup. */
+  public static final Duration TEST_RESOURCE_TIME_TO_LIVE = Duration.ofHours(1);
+
+  private final CrlConfiguration crlConfig;
+
+  private final CloudResourceManagerCow crlResourceManagerCow;
+  private final CloudBillingClientCow crlBillingClientCow;
+  private final ServiceUsageCow crlServiceUsageCow;
+
+  @Autowired
+  public CrlService(CrlConfiguration crlConfig) {
+    this.crlConfig = crlConfig;
+
+    ClientConfig clientConfig;
+    if (crlConfig.useCrl()) {
+      var creds = buildCredentials();
+      clientConfig = buildClientConfig();
+      try {
+        this.crlResourceManagerCow = CloudResourceManagerCow.create(clientConfig, creds);
+        this.crlBillingClientCow = new CloudBillingClientCow(clientConfig, creds);
+        this.crlServiceUsageCow = ServiceUsageCow.create(clientConfig, creds);
+
+      } catch (GeneralSecurityException | IOException e) {
+        throw new CrlInternalException("Error creating resource manager wrapper", e);
+      }
+    } else {
+      clientConfig = null;
+      crlResourceManagerCow = null;
+      crlBillingClientCow = null;
+      crlServiceUsageCow = null;
+    }
+  }
+
+  /**
+   * @return CRL {@link CloudResourceManagerCow} which wrappers Google Cloud Resource Manager API
+   */
+  public CloudResourceManagerCow cloudResourceManagerCow() {
+    assertCrlInUse();
+    return crlResourceManagerCow;
+  }
+
+  /** The CRL {@link CloudBillingClientCow} which wrappers Google Billing API. */
```

Comment:
nit: use the same comment style for all of these Cows?

Reply:
I inherited those comments from the existing implementation. Happy to clean them up!